src/lib/executeSourceTransfers.js

```diff
@@ -58,15 +59,20 @@ function * executeSourceTransfers (sourceTransfers, destinationTransfers, relate
     return
   }
 
-  for (let sourceTransfer of sourceTransfers) {
-    log.debug('Requesting fulfillment of source transfer: ' + sourceTransfer.id + ' (fulfillment: ' + JSON.stringify(conditionFulfillment) + ')')
+  for (let destinationTransfer of destinationTransfers) {
+    const destinationDebitMemo = destinationTransfer.debits[0].memo || {}
+    const sourceTransferLedger = destinationDebitMemo.source_transfer_ledger
+    const sourceTransferID = destinationDebitMemo.source_transfer_id
+    validate('Iri', sourceTransferLedger)
+    validate('Iri', sourceTransferID)
+
+    log.debug('Requesting fulfillment of source transfer: ' + sourceTransferID + ' (fulfillment: ' + JSON.stringify(conditionFulfillment) + ')')
     // TODO check the timestamp on the response from the ledger against
     // the transfer's expiry date
     // See https://github.com/interledger/five-bells-ledger/issues/149
-    yield ledgers.putTransferFulfillment(sourceTransfer, conditionFulfillment)
-    if (sourceTransfer.state !== 'executed') {
-      log.error('Attempted to execute source transfer but it was unsucessful')
-      log.debug(sourceTransfer)
+    const sourceTransferState = yield ledgers.putTransferFulfillment(sourceTransferLedger, sourceTransferID, conditionFulfillment)
```

Comment:
This looks like it doesn't match the function signature of `putTransferFulfillment`, which expects just `transferID, execution_condition_fulfillment`